gadgets/traceloop/go/program.go

```diff
@@ -0,0 +1,649 @@
+	for {
+		record, err := reader.perfReader.Read()
+		if err != nil {
+			if errors.Is(err, os.ErrDeadlineExceeded) {
+				break
+			} else {
+				return nil, err
+			}
+		}
+		records = append(records, record)
+	}
+
+	err = reader.perfReader.Resume()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, record := range records {
+		size := len(record)
+
+		var sysEvent *traceloopSyscallEventT
+		var sysEventCont *traceloopSyscallEventContT
+
+		switch uintptr(size) {
+		case alignSize(unsafe.Sizeof(*sysEvent)):
+			sysEvent = (*traceloopSyscallEventT)(unsafe.Pointer(&record[0]))
+
+			event := &syscallEvent{
+				bootTimestamp:      sysEvent.BootTimestamp,
+				monotonicTimestamp: sysEvent.MonotonicTimestamp,
+				typ:                sysEvent.Typ,
+				contNr:             sysEvent.ContNr,
+				cpu:                sysEvent.Cpu,
+				id:                 sysEvent.Id,
+				pid:                sysEvent.Pid,
+				comm:               fromCString(sysEvent.Comm[:]),
+				mountNsID:          reader.mntnsID,
+			}
+
+			var typeMap *map[uint64][]*syscallEvent
+			switch event.typ {
+			case syscallEventTypeEnter:
+				event.args = make([]uint64, syscallArgs)
+				for i := uint8(0); i < syscallArgs; i++ {
+					event.args[i] = sysEvent.Args[i]
+				}
+
+				typeMap = &syscallEnterEventsMap
+			case syscallEventTypeExit:
+				event.retval = sysEvent.Args[0]
+
+				typeMap = &syscallExitEventsMap
+			default:
+				// Rather than returning an error, we skip this event.
+				api.Debugf("type %d is not a valid type for syscallEvent, received data are: %v", event.typ, record)
+
+				continue
+			}
+
+			if _, ok := (*typeMap)[event.monotonicTimestamp]; !ok {
+				(*typeMap)[event.monotonicTimestamp] = make([]*syscallEvent, 0)
+			}
+
+			(*typeMap)[event.monotonicTimestamp] = append((*typeMap)[event.monotonicTimestamp], event)
+		case alignSize(unsafe.Sizeof(*sysEventCont)):
+			sysEventCont = (*traceloopSyscallEventContT)(unsafe.Pointer(&record[0]))
+
+			event := &syscallEventContinued{
+				monotonicTimestamp: sysEventCont.MonotonicTimestamp,
+				index:              sysEventCont.Index,
+			}
+
+			if sysEventCont.Failed != 0 {
+				event.param = "(Failed to dereference pointer)"
+			} else if sysEventCont.Length == useNullByteLength {
+				// 0 byte at [C.PARAM_LENGTH - 1] is enforced in BPF code
+				event.param = fromCString(sysEventCont.Param[:])
+			} else {
+				event.param = fromCStringN(sysEventCont.Param[:], int(sysEventCont.Length))
+			}
+
+			// Remove all non unicode character from the string.
+			event.param = strconv.Quote(event.param)
+
+			_, ok := syscallContinuedEventsMap[event.monotonicTimestamp]
+			if !ok {
+				// Just create a 0 elements slice for the moment, the ContNr will be
+				// checked later.
+				syscallContinuedEventsMap[event.monotonicTimestamp] = make([]*syscallEventContinued, 0)
+			}
+
+			syscallContinuedEventsMap[event.monotonicTimestamp] = append(syscallContinuedEventsMap[event.monotonicTimestamp], event)
```

Comment:
```suggestion
			syscallContinuedEventsMap[event.monotonicTimestamp] = append(syscallContinuedEventsMap[event.monotonicTimestamp], event)
```

IIUC this should work fine. Same in similar places.